sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -59,8 +62,58 @@ public void enableSessionCapturing(boolean sessionCapturingOverride) {
      *
      * @return the session capturing override
      */
-    public boolean isSessionCapturingOverride() {
-        return this.sessionCapturingOverride;
+    public boolean isSessionCapturingOverrideEnabled() {
+        return this.sessionCapturingOverrideEnabled;
+    }
+
+    /**
+     * Enables connections sharing across multiple Cosmos Clients. The default is false.
+     *
+     *
+     * <pre>
+     * {@code
+     * ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+     * getConnectionPolicy.getConnectionMode(ConnectionMode.DIRECT);
+     * CosmosAsyncClient client1 = new CosmosAsyncClient.cosmosClientBuilder()
+     *         .setEndpoint(serviceEndpoint1)
+     *         .setKey(key1)
+     *         .setConnectionPolicy(connectionPolicy)
+     *         .setConsistencyLevel(ConsistencyLevel.SESSION)
+     *         .setConnectionSharingAcrossClientsEnabled(true)
+     *         .buildAsyncClient();
+     *
+     * CosmosAsyncClient client2 = new CosmosAsyncClient.cosmosClientBuilder()
+     *         .setEndpoint(serviceEndpoint2)
+     *         .setKey(key2)
+     *         .setConnectionPolicy(connectionPolicy)
+     *         .setConsistencyLevel(ConsistencyLevel.SESSION)
+     *         .setConnectionSharingAcrossClientsEnabled(true)
+     *         .buildAsyncClient();
+     *
+     * // when configured this way client1 and client2 will share connections when possible.
+     * }
+     * </pre>
+     *
+     * When you have multiple instances of Cosmos Client in the same JVM interacting to multiple Cosmos accounts,
+     * enabling this allows connection sharing in Direct mode if possible between instances of Cosmos Client.
+     * @param connectionSharingAcrossClientsEnabled connection sharing
+     * @return current cosmosClientBuilder
+     */
+    public CosmosClientBuilder setConnectionSharingAcrossClientsEnabled(boolean connectionSharingAcrossClientsEnabled) {
```

Comment:
In SetConnectionSharingAcrossClientsEnabled  "acrossClients" sounds little verbose , how about setClientsConnectionSharingEnabled()/ isClientsConnectionSharingEnabled().

Reply:
resolving comment here , offline conversation going on